source/bsd/kqueue_event_loop.c

```diff
@@ -356,6 +360,7 @@ static int s_run(struct aws_event_loop *event_loop) {
         AWS_LOGF_FATAL(AWS_LS_IO_EVENT_LOOP, "id=%p: thread creation failed.", (void *)event_loop);
         goto clean_up;
     }
+    aws_atomic_store_int(&impl->thread_id, aws_thread_get_id(&impl->thread));
```

Comment:
we should set the thread_id from the beginning of `s_event_thread_main()`, not from here